proxy/src/console/messages.rs

```diff
@@ -0,0 +1,190 @@
+use serde::Deserialize;
+use std::fmt;
+
+/// Generic error response with human-readable description.
+/// Note that we can't always present it to user as is.
+#[derive(Debug, Deserialize)]
+pub struct ConsoleError {
+    pub error: Box<str>,
```

Comment:
Is this code hand written or autogenerated? Are there code generation tools for OpenAPI spec 3.0 in rust? There is at least [openapi-generator](https://www.twilio.com/docs/openapi/generating-a-rust-client-for-twilios-api) which is not a good solution, being in Java. But at least it allows to not write tests on JSON deserialization.

Reply:
This is hand written. There are several reasons for that:

* (Historical reasons) That's how it's been done here for quite a while, and it's not that bad, given that:
* The spec is incomplete! It doesn't cover http status codes or headers like `X-Request-ID`.
* The spec lives in a separate private repo, there's no cross-CI checks to validate the generated code.
* It's not like we're doing a lot of different requests here, and handwritten code gives you you more control.

Reply:
Right, I got you.